### ISSUE TITLE
there are some errors while translating tinyexpr c library

### ISSUE THREAD

**Issue by piaoger** [immunant/c2rust#29]
How to reproduce:
1. go to https://c2rust.com/. I tried to use the online demo because establishing environment is boring.
2. copy tinyexpr.h and tinyexpr.c as c source code
   https://raw.githubusercontent.com/codeplea/tinyexpr/master/tinyexpr.h
   https://raw.githubusercontent.com/codeplea/tinyexpr/master/tinyexpr.c
3. run "Translate"
4. cargo new a test app and copy the result .rs to the project
5. compile and found below errors
  
<img width="639" alt="image" src="https://user-images.githubusercontent.com/179429/46904801-7f811500-cf1c-11e8-8a50-f0c91cb38a56.png">

**Comment by glguy** [comment]
@piaoger This was actually fixed in 9de4cd645e055b6b525eda10d1c4cbe6cb319a4c . Could you try updating your repository and seeing if that resolves any of the errors you were encountering?

**Comment by piaoger** [comment]
It seems that “https://c2rust.com/” is not updated now. I will try again later

**Comment by TheDan64** [comment]
The offset issue should indeed be fixed on master.

This seems to reproduce the mutability issue:
```c
void new_expr(const char *parameters[]) {}

#define NEW_EXPR(...) new_expr((const char*[]){__VA_ARGS__})

int main(void) {
    char *a = "asd";
    NEW_EXPR(a);
}
```

Maybe we're missing some implicit mutability cast here?

**Comment by glguy** [comment]
Yeah, the const cast is in the AST, but we're omitting it. I've stripped out even more of the test case:

```cpp
void isolated(void) {
    char *a;
    const char **p = (const char*[]){a};
}
```

```rust
pub unsafe extern "C" fn isolated() -> () {
    let mut a: *mut libc::c_char = 0 as *mut libc::c_char;
    let mut p: *mut *const libc::c_char = [a].as_mut_ptr();
}
```

**Comment by glguy** [comment]
@piaoger c2rust.com is updated, but we still need to fix this const issue.

**Comment by piaoger** [comment]
@glguy  I just checked and found the resultant rust file can be built successfully. Thanks for your help.

P.S.
"https://c2rust.com/" is great for tiny projects.Do we have a pre-built releases for Linux or Mac? Or the demo app can support uploading package. Establishing c2rust env seems not very easy. It's will be wonderful and helpful if there is something as "cargo install c2rust"  :)

**Comment by TheDan64** [comment]
We are investigating the possibility of release binaries and/or cargo install at this time. We would certainly like to see it happen.

**Comment by piaoger** [comment]
> We are investigating the possibility of release binaries and/or cargo install at this time. We would certainly like to see it happen.
Wow, that should be great for rust community 👍
